app/src/main/java/org/openedx/app/analytics/SegmentAnalytics.kt

```diff
@@ -12,18 +17,46 @@ import com.segment.analytics.kotlin.core.Analytics as SegmentTracker
 class SegmentAnalytics(context: Context, config: Config) : Analytics {
 
     private val logger = Logger(TAG)
-    private var tracker: SegmentTracker
 
-    init {
-        // Create an analytics client with the given application context and Segment write key.
-        tracker = SegmentAnalyticsBuilder(config.getSegmentConfig().segmentWriteKey, context) {
+    // Create an analytics client with the given application context and Segment write key.
+    private var tracker: SegmentTracker =
+        SegmentAnalyticsBuilder(config.getSegmentConfig().segmentWriteKey, context) {
             // Automatically track Lifecycle events
             trackApplicationLifecycleEvents = true
             flushAt = 20
             flushInterval = 30
         }
+
+    init {
         if (config.getFirebaseConfig().isSegmentAnalyticsSource()) {
             tracker.add(plugin = FirebaseDestination(context = context))
+
+            tracker.find(FirebaseDestination::class)?.add(object : EventPlugin {
```

Comment:
deserve doc string, why need to use the find function, cuz otherwise the app leads to a crash.